presto-redis/src/main/java/com/facebook/presto/redis/RedisJedisManager.java

```diff
@@ -76,14 +98,117 @@ public JedisPool getJedisPool(HostAddress host)
         return jedisPoolCache.getUnchecked(host);
     }
 
-    private JedisPool createConsumer(HostAddress host)
+    /**
+     * Creates a new JedisPool for the specified host.
+     * Chooses between TLS or non-TLS configuration based on redisConnectorConfig.
+     */
+    private JedisPool createJedisPool(HostAddress host)
+    {
+        if (redisConnectorConfig.isTlsEnabled()) {
+            SSLContext sslContext = createSslContext(loadTrustStore());
+            return createTlsJedisPool(host, sslContext);
+        }
+        else {
+            return createNonTlsJedisPool(host);
```

Comment:
Your `createTlsJedisPool` & `createNonTlsJedisPool` internally uses same constructor just different values. We can modify it and keep just one method instead of repeating